### ISSUE TITLE
MultiChart tooltip doesn't format header

### ISSUE THREAD

**Issue by toyoaki** [novus/nvd3#999]
I have a graph that displays formated dates on the X axis ('%Y %b') but when I hover the mouse, the tooltip doens't display the header with the date formated. By adding the code:

```javascript
.headerFormatter(function(d, i) {
   return xAxis.tickFormat()(d, i);
})
```

To the tooltips, the problem got solved.

Thank you for making such a good graphics lib.

**Comment by ranjithsiji** [comment]
@toyoaki  Thank you for posting this great tip. Saved my life.

**Comment by toyoaki** [comment]
Glad I could help! =]

**Comment by liquidpele** [comment]
I think this might be fixed in master now

**Comment by ger-morab-dct** [comment]
Yes, this issue was fixed for me when I updated to 1.8.3. Thanks.

**Comment by inolasco** [comment]
I have a problem with this, it's overriding my custom `headerFormatter` function. My function gets set initially, but then when the tooltip is about to be displayed, this function gets called instead.

**Comment by akhilravipati97** [comment]
I've had this issue with nvd3's multichart interactiveGuideline. And it is the same behavior as described by @inolasco. Quite a lot of tooltip events and interactiveGuideline events for tootltips **ignore** any already overridden headerFormatters by the user and simply set it to the xAxis Formatter. 

**Surprisingly**, at least the interactiveGuideline, prioritizes any overridden valueFormatters. Code:

```
interactiveLayer.tooltip
        .headerFormatter(function(d, i) {
            return xAxis.tickFormat()(d, i);
        })
        .valueFormatter(interactiveLayer.tooltip.valueFormatter() || defaultValueFormatter)
        .data({
            value: chart.x()( singlePoint,pointIndex ),
            index: pointIndex,
            series: allData
        })();
```

So, I guess, **one fix** is to replicate the code for valueFormatter and make headerFormatter use any already overridden value:
```
.headerFormatter(interactiveLayer.tooltip.headerFormatter() || function(d, i) {
      return xAxis.tickFormat()(d, i);
  })
```

**But** that would mean you have to change the library code, minify it and distribute separately by yourself. And God knows how many other places in the library this has to be patched.


So, as a **quick hack**, disable headerFormatter from getting overridden once the custom headerFormatter is set:
```
// set new formatter
const newHeaderFormatter = function(d) { .... };
chart.interactiveLayer.tooltip.headerFormatter(newHeaderFormatter);

// don't let it get overridden by making the setter ignore arguments and simply return parent object (nvd3's builder pattern)
chart.interactiveLayer.tooltip.headerFormatter = (d: any)=>{return chart.interactiveLayer.tooltip;};
```

This hack should probably also apply to a chart's tooltip too (without the interactiveLayer). I guess the return value should instead by `chart.tooltip`
